### ISSUE TITLE
Allow anything that evaluates to string to be an "include" path

### ISSUE THREAD

**Issue by fkooman** [vimeo/psalm#932]
```php
<?php
$autoloadPath = '/foo/bar/baz/autoload.php';
require_once $autoloadPath;
```

This triggers the `UnresolvableInclude` error, it seems `IncludeChecker` supports some different ways to figure out includes, but not this one. It supports strings, calls to `dirname` and consts like `__DIR__` and `__FILE__`, but *not* variables.

Proposal: accept everything that is either a string, or evaluates to a string.

**Comment by fkooman** [comment]
> or evaluates to a string

My `require_once` looks like this, but maybe that is not such a good idea and I should change it?

```php
<?php
$baseDir = dirname(__DIR__);
require_once sprintf('%s/vendor/autoload.php', $baseDir);
```

**Comment by muglug** [comment]
Psalm’s just saying it couldn’t resolve the expression to a single file path. Here that’s because it doesn’t execute `sprintf` statements as part of its include file interpolation.
